### ISSUE TITLE
[Bug] Notification not working on linux

### ISSUE THREAD

**Issue by ZXSwire3** [sgtaziz/WebMessage#181]
I tried using the .AppImage file and installing it through snap, but both resulted in no notifications in general. Notifications would not appear in my system notifications at all, and worst of all I would not get a notification sound with either the default notification sound or with the custom sound.

**Comment by CSab6482** [comment]
Does [this pull request](https://github.com/sgtaziz/WebMessage/pull/185) fix the issue?

**Comment by ZXSwire3** [comment]
yes and no. There was a lot of trouble trying to build that version of webmessage because it seems like one of the dependencies seems to just be gone, but luckily I had a version of it on my desktop from when I was trying to build it before so I was able to get it to work after a lot of trial and error. Anyways, the notifications work when I compile it for hot-reloads, but not when built for production. I'm not really sure what the issue is, maybe it's just something with how I have it configured or maybe the way that it builds the .AppImage and snap files is weird and it doesn't fully support notifications.

**Comment by ZXSwire3** [comment]
> How can I build this version?

It's kind of complicated, I believe that some of the dependencies for this project has just disappeared. I think I still have the dependency that was removed so I can just upload it to github so that you're able to clone it and build it. Have you installed npm and yarn? If you have both of those then all you'll need is that missing dependency that I will upload when I get a chance. I'll reply to this message with the link to it when I upload it.

> is your notification fixed now?

yes and no, if all you want is for a notification to pop up in your notification tray then yes, but if you want a sound notification then it won't work still.

> have you tried SMserver by any chance?

I have tried it, I personally didn't like it as much as WebMessage, but it's really a personal choice. I didn't like that I had to have a browser open to use it and I just thought that WebMessage looked better. I think that SMServer might be a better choice over all because I think it has more features that WebMessage, but I have pretty much everything that I want from WebMessage so I personally won't be changing unless SMServer gets more updates. It seems like both are not actively in development so whichever gets more features sooner I'd probably use.

> I noticed the ip address to connect to tweak app in my iphone changes so every time that happens I have to try to resync between my laptop and iphone with renewed ip. Is there any way to use static one?


You can actually set a static ip on your router for that specific device that you are using as a host. I did this because I was also getting tired of having to reconnect the client with the host.

**Comment by ZXSwire3** [comment]
> When I tried to build his original version

What did you run? Have you tried using this one? [https://github.com/shiv213/WebMessage](url) 

> As for SMserver, I couldn't connect to the address/port that I set to for some reason. Do you know if this app's notification works?

Did you have WebMessage daemon running at the same time on your device? I believe that when I was trying to set up one or the other, I would have to remove the tweak or at least disable it with something like iCleaner. I believe that the notifications would work for it since it's just runs in the browser, as long as your browser is able to send notifications.

>  when you said pop up, did you mean a notification with actual message preview?

I believe that it did show the actual message, but I only used it in xfce so I'm not sure how it will respond in kde. It for sure does show a pop up though.

> If I want to set a static ip on my router. I can just set it for this laptop but not the rest of devices at home?

You would want to set the device that is running the WebMessage tweak to have a static ip. It's different for every router so you would have to either look up how to do it on your router or just dig around in your router settings to find it. For example, on my router it's under `Advanced - DHCP Server - Static Leases` but it could be anywhere in the settings on yours. All you have to do once you navigate to that area is find the mac address of the device that has the tweak running and then put it into your router and tell it what static ip you want it to have. It will then have that ip on your local network for as long as you have that setup, but it's just that device that will have its ip as that. For example, I have mine set to be `10.1.1.114`

**Comment by shawnk0723** [comment]
> > When I tried to build his original version
> 
> What did you run? Have you tried using this one? [https://github.com/shiv213/WebMessage](url)
> 
> > As for SMserver, I couldn't connect to the address/port that I set to for some reason. Do you know if this app's notification works?
> 
> Did you have WebMessage daemon running at the same time on your device? I believe that when I was trying to set up one or the other, I would have to remove the tweak or at least disable it with something like iCleaner. I believe that the notifications would work for it since it's just runs in the browser, as long as your browser is able to send notifications.
> 
> > when you said pop up, did you mean a notification with actual message preview?
> 
> I believe that it did show the actual message, but I only used it in xfce so I'm not sure how it will respond in kde. It for sure does show a pop up though.
> 
> > If I want to set a static ip on my router. I can just set it for this laptop but not the rest of devices at home?
> 
> You would want to set the device that is running the WebMessage tweak to have a static ip. It's different for every router so you would have to either look up how to do it on your router or just dig around in your router settings to find it. For example, on my router it's under `Advanced - DHCP Server - Static Leases` but it could be anywhere in the settings on yours. All you have to do once you navigate to that area is find the mac address of the device that has the tweak running and then put it into your router and tell it what static ip you want it to have. It will then have that ip on your local network for as long as you have that setup, but it's just that device that will have its ip as that. For example, I have mine set to be `10.1.1.114`

Hi ZXSwire3,

I'm so sorry for the late reply. There was a family emergency for the past few days. Thanks a lot for all the tips. 
Unfortunately, I'm currently away from my laptop. Once I get a chance to try them then I'll definitely keep you posted. Thanks a lot and have a wonderful weekend! :) 

Best

**Comment by shawnk0723** [comment]
> What did you run? Have you tried using this one? [https://github.com/shiv213/WebMessage](url)
> 
> >

What did you run? Have you tried using this one? [https://github.com/shiv213/WebMessage](https://github.com/sgtaziz/WebMessage/issues/url)

yes, that is the version I tried to build but It failed. I would appreciate it very much if you can send me the link to your version that includes all the dependencies to build this.
Thank you in advance. :)

**Comment by shawnk0723** [comment]
> > How can I build this version?
> 
> It's kind of complicated, I believe that some of the dependencies for this project has just disappeared. I think I still have the dependency that was removed so I can just upload it to github so that you're able to clone it and build it. Have you installed npm and yarn? If you have both of those then all you'll need is that missing dependency that I will upload when I get a chance. I'll reply to this message with the link to it when I upload it.
> 
> > is your notification fixed now?
> 
> yes and no, if all you want is for a notification to pop up in your notification tray then yes, but if you want a sound notification then it won't work still.
> 
> > have you tried SMserver by any chance?
> 
> I have tried it, I personally didn't like it as much as WebMessage, but it's really a personal choice. I didn't like that I had to have a browser open to use it and I just thought that WebMessage looked better. I think that SMServer might be a better choice over all because I think it has more features that WebMessage, but I have pretty much everything that I want from WebMessage so I personally won't be changing unless SMServer gets more updates. It seems like both are not actively in development so whichever gets more features sooner I'd probably use.
> 
> > I noticed the ip address to connect to tweak app in my iphone changes so every time that happens I have to try to resync between my laptop and iphone with renewed ip. Is there any way to use static one?
> 
> You can actually set a static ip on your router for that specific device that you are using as a host. I did this because I was also getting tired of having to reconnect the client with the host.

I tried smserver after removing tweak but it still doesn't connect to the address/port I set it to.

**Comment by shawnk0723** [comment]
> Does [this pull request](https://github.com/sgtaziz/WebMessage/pull/185) fix the issue?

Hey @CSab6482, 

Were you able to fix this issue? Last time I tried that version, I got some dependencies related error. Were you able to build his code ok? I tried to follow up with @ZXSwire3 but he hasn't been replying to my last messages. It's pretty frustrating without notification.

**Comment by ZXSwire3** [comment]
> > Does [this pull request](https://github.com/sgtaziz/WebMessage/pull/185) fix the issue?
> 
> Hey @CSab6482,
> 
> Were you able to fix this issue? Last time I tried that version, I got some dependencies related error. Were you able to build his code ok? I tried to follow up with @ZXSwire3 but he hasn't been replying to my last messages. It's pretty frustrating without notification.

I'm so sorry, life got busy and I just forgot about it. I think I might have found the backup of it. [This](https://mega.nz/file/T01iUCQb#EQ2Bo1yUCtSYRbjf8lvEqLMwZXGl3YSp9fvVT_Zi5a0) should be the correct one, I would test it but I don't currently have a linux machine anymore and it seems like Windows is just super messed up when it comes to building things. Let me know if it works, otherwise I might not be able to help out.

**Comment by shawnk0723** [comment]
> > > Does [this pull request](https://github.com/sgtaziz/WebMessage/pull/185) fix the issue?
> > 
> > 
> > Hey @CSab6482,
> > Were you able to fix this issue? Last time I tried that version, I got some dependencies related error. Were you able to build his code ok? I tried to follow up with @ZXSwire3 but he hasn't been replying to my last messages. It's pretty frustrating without notification.
> 
> I'm so sorry, life got busy and I just forgot about it. I think I might have found the backup of it. [This](https://mega.nz/file/T01iUCQb#EQ2Bo1yUCtSYRbjf8lvEqLMwZXGl3YSp9fvVT_Zi5a0) should be the correct one, I would test it but I don't currently have a linux machine anymore and it seems like Windows is just super messed up when it comes to building things. Let me know if it works, otherwise I might not be able to help out.

@ZXSwire3, 

Thank you for the files but how can I go about installing it? Also, is this same version as the following? https://github.com/sgtaziz/WebMessage/pull/185

Last time I tried to build it myself, I ended up giving it up because it kept on giving me dependencies error. I'm not sure if I was doing something wrong but I just wanna make sure I do this correctly this time. 
FYI, I installed my current webMessage 0.7.0 version using appimage. 
I would appreciate it very much if you can briefly walk me through this. Thank you in advance. 

P.S.

BTW, I was able to try smserver as well as static address for tweak thanks to your tips. :) I liked how smserver's notification worked without extra configurations. However, smserver's notification only worked for me when it was in https mode but it was either not getting new messages or slow to update new messages in that mode. In http, everything worked perfect except for the notification. In addition, I didn't like how smserver app had to be kept open on my phone in order to keep the server running in the background (I'm sure there are some tricks to hide it)   
how was your experience with it when you tried it? what did you like/dislike compare to webmessage?
overall, I prefer webmessage. It will be just perfect if I can get notification working in kde.

**Comment by ZXSwire3** [comment]
> Thank you for the files but how can I go about installing it? Also, is this same version as the following? https://github.com/sgtaziz/WebMessage/pull/185
I can't remember to be sure, I'm pretty sure it is because I believe that I downloaded that version and just put the dependencies into it, but if you want to be safe then you can just clone that repo and then copy all the files from https://github.com/sgtaziz/WebMessage/pull/185 and put them into the one that I provided.

As for installing you just want to navigate to the folder in your terminal and then type `yarn serve`. I believe that you don't have to do `yarn install` because the dependencies should already be there. The only issue with doing it like this is that you have to leave that terminal open because that's where it's running the electron app. From what I remember if you try it build the appimage or snap file it won't actually work.

**Comment by shawnk0723** [comment]
> > Thank you for the files but how can I go about installing it? Also, is this same version as the following? #185
> 
> I can't remember to be sure, I'm pretty sure it is because I believe that I downloaded that version and just put the dependencies into it, but if you want to be safe then you can just clone that repo and then copy all the files from #185 and put them into the one that I provided.
> 
> As for installing you just want to navigate to the folder in your terminal and then type `yarn serve`. I believe that you don't have to do `yarn install` because the dependencies should already be there. The only issue with doing it like this is that you have to leave that terminal open because that's where it's running the electron app. From what I remember if you try it build the appimage or snap file it won't actually work.

@ZXSwire3 

Thanks a lot for the quick reply. I tried yarn serve but it gives me an error " ERROR  TypeError: transpileDependencies.map is not a function". I'm not familiar with vue but brief googling says to update the vue to 5.0.0. Any idea what might be the  cause of this?

**Comment by shawnk0723** [comment]
@ZXSwire3 
Another thing I noticed was  there's win-unpacked inside of build folder instead of linux-unpacked. Have you been using it in your windows system? in that case, I'll have to rebuild it. I've already tried both serve and build but it returns the above error about transpileDependencies.map. I also tried to change the vue version to 5 from package.json then npm i but it kept on giving me different dependencies errors. I'm not sure if I'm doing this correctly. Any help would be greatly appreciated.

**Comment by ZXSwire3** [comment]
I'm not too familiar with vue either, I think that I might have tried doing `yarn install` in the folder. If it doesn't work I'm not really sure what else that I can do to help. I'll try checking on the linux machine that I was able to build it on and see if it's the same as the one that uploaded.

**Comment by ZXSwire3** [comment]
ok, I just got on that machine and I was able to build it, [this one](https://mega.nz/file/yl0yXJwL#5pqYU1_yP1056R6jJ4PzAAc1Uj6sjTjncBB6lTQYrQ4) is for sure the correct one. Let me know if it works

**Comment by shawnk0723** [comment]
> I'm not too familiar with vue either, I think that I might have tried doing `yarn install` in the folder. If it doesn't work I'm not really sure what else that I can do to help. I'll try checking on the linux machine that I was able to build it on and see if it's the same as the one that uploaded.



> ok, I just got on that machine and I was able to build it, [this one](https://mega.nz/file/yl0yXJwL#5pqYU1_yP1056R6jJ4PzAAc1Uj6sjTjncBB6lTQYrQ4) is for sure the correct one. Let me know if it works

@ZXSwire3 

LOL, perfect timing. I was just replying to your previous email. thanks a lot! I'll keep you posted after trying this version. :)

**Comment by shawnk0723** [comment]
@ZXSwire3 

Yes!! Finally it's working! what did you do differently? so you don't remember where you got this version from? this is perfect!! makes me really wonder how you were able to build this in case this breaks again and I have to do it myself later. so it was just that one dependency that you had to add on your own? do you remember what it was called? only error I kept on getting was "TypeError: transpileDependencies.map is not a function". 

Anyways, thank you so much for all your help! :)
